test/unit/provisioning_template_test.rb

```diff
@@ -85,10 +85,26 @@ class ProvisioningTemplateTest < ActiveSupport::TestCase
     assert FactoryGirl.create(:provisioning_template, :locked => true)
   end
 
-  test "should not edit a locked template" do
-    tmplt = templates(:locked)
-    tmplt.name = "something else"
-    refute_valid tmplt, :base, /is locked/
+  context 'locked templates outside of rake' do
+    setup do
+      Foreman.stubs(:in_rake?).returns(false)
```

Comment:
Nitpick, prefer `.expects` just to ensure we're exercising the code as we expect.  You'd also need to add `.at_least_once` to allow it to call in_rake? twice.